Libraries/Animated/src/AnimatedImplementation.js

```diff
@@ -1962,76 +1995,123 @@ var spring = function(
-  return maybeVectorAnim(value, config, spring) || {
-    start: function(callback?: ?EndCallback): void {
-      callback = _combineCallbacks(callback, config);
-      var singleValue: any = value;
-      var singleConfig: any = config;
+  var start = function(
+    animatedValue: AnimatedValue | AnimatedValueXY,
+    configuration: SpringAnimationConfig,
+    callback?: ?EndCallback): void {
+      callback = _combineCallbacks(callback, configuration);
+      var singleValue: any = animatedValue;
+      var singleConfig: any = configuration;
       singleValue.stopTracking();
-      if (config.toValue instanceof Animated) {
+      if (configuration.toValue instanceof Animated) {
         singleValue.track(new AnimatedTracking(
           singleValue,
-          config.toValue,
+          configuration.toValue,
           SpringAnimation,
           singleConfig,
           callback
         ));
       } else {
         singleValue.animate(new SpringAnimation(singleConfig), callback);
       }
+  };
+  return maybeVectorAnim(value, config, spring) || {
+    start: function(callback?: ?EndCallback): void {
+      start(value, config, callback);
     },
 
     stop: function(): void {
       value.stopAnimation();
     },
+
+    reset: function(): void {
+      value.resetAnimation();
+    },
+
+    startNativeLoop: function(iterations: number): void {
+      config.numLoops = iterations;
+      start(value, config);
+    }
   };
 };
 
 var timing = function(
   value: AnimatedValue | AnimatedValueXY,
   config: TimingAnimationConfig,
 ): CompositeAnimation {
-  return maybeVectorAnim(value, config, timing) || {
-    start: function(callback?: ?EndCallback): void {
-      callback = _combineCallbacks(callback, config);
-      var singleValue: any = value;
-      var singleConfig: any = config;
+  var start = function(
+    animatedValue: AnimatedValue | AnimatedValueXY,
+    configuration: TimingAnimationConfig,
+    callback?: ?EndCallback): void {
+      callback = _combineCallbacks(callback, configuration);
+      var singleValue: any = animatedValue;
+      var singleConfig: any = configuration;
       singleValue.stopTracking();
-      if (config.toValue instanceof Animated) {
+      if (configuration.toValue instanceof Animated) {
         singleValue.track(new AnimatedTracking(
           singleValue,
-          config.toValue,
+          configuration.toValue,
           TimingAnimation,
           singleConfig,
           callback
         ));
       } else {
         singleValue.animate(new TimingAnimation(singleConfig), callback);
       }
+  }
+
+  return maybeVectorAnim(value, config, timing) || {
+    start: function(callback?: ?EndCallback): void {
+      start(value, config, callback);
     },
 
     stop: function(): void {
       value.stopAnimation();
     },
+
+    reset: function(): void {
+      value.resetAnimation();
+    },
+
+    startNativeLoop: function(iterations: number): void {
+      config.numLoops = iterations;
```

Comment:
Let's avoid mutating the config object, in case it gets reused for multiple animations, `{ ...config, iterations }`.